lib/modules/manager/composer/artifacts.ts

```diff
@@ -62,7 +62,12 @@ function getAuthJson(): string | null {
     .findAll({ hostType: PackagistDatasource.id })
     ?.forEach((hostRule) => {
       const { resolvedHost, username, password, token } = hostRule;
-      if (resolvedHost && username && password) {
+      if (resolvedHost === 'github.com' && token) {
+        // Allow the use of a GitHub access token for packagist only.
+        authJson['github-oauth'] = {
+          'github.com': token.replace('x-access-token:', ''),
```

Comment:
Does this "oauth" actually work with all types of GitHub tokens? Eg including PAT and App token

Reply:
Good question, I tested with a PAT

Reply:
https://getcomposer.org/doc/articles/authentication-for-private-packages.md#github-oauth

we should check that only a PAT is used for this

Reply:
Do we have any mean to verifiy that a token is actually a pat ? the `ghp_` prefix ?

Reply:
Just answered myself https://github.blog/2021-04-05-behind-githubs-new-authentication-token-formats/

so  `ghp` is a personnal access token and `ghs` is a server-to-server token.

Now that I see it it's so obvious

> Support for password authentication was removed on August 13, 2021.  Please use a personal access token instead. 

So the default renovate application token cannot be used for x-oauth-basic in git clone (by compser).

I will update the pr to take the git-tags rule first to avoid using the default github rule whish have a `ghs` token by default.

@viceice